src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -109,7 +109,7 @@ def config_dashboard(self, daemon_descrs: List[DaemonDescription]):
         """Config dashboard settings."""
         raise NotImplementedError()
 
-    def get_active_daemon(self, daemon_descrs: List[DaemonDescription]) -> DaemonDescription:
+    def get_active_daemon(self, daemon_descrs: List[DaemonDescription]) -> Optional[DaemonDescription]:
```

Comment:
Please also have a look at 

https://github.com/ceph/ceph/blob/c38309f0c2f3421416c35ea5eab793426be39101/src/pybind/mgr/cephadm/services/monitoring.py#L62-L64

Reply:
Ended up reverting the "Optional" on the return type of this function in the process of trying to not limit to mgr and mds in other places.